src/libre/ac.c

```diff
@@ -126,7 +135,7 @@ trie_create(void)
 }
 
 struct trie_graph *
-trie_add_word(struct trie_graph *g, const char *w, size_t n)
+trie_add_word(struct trie_graph *g, const char *w, size_t n, const fsm_end_id_t *endid)
```

Comment:
i don't think `endid` needs to be a pointer here

Reply:
It's a pointer because it's an optional argument and can be NULL, coming from `re_strings_add_str`. What I'm going for here is `Option<fsm_end_id_t>` in Rust terms. Instead of using a pointer it could be two versions of the same function call, or have a `ENDID_NONE` constant (probably `((fsm_end_id_t)-1)`), would you prefer one of those?